docs/source_values.rst

```diff
@@ -178,12 +185,106 @@ In this example, the LED is lit only when the button is not pressed:
 
 Some tools combine the values of multiple sources:
 
-.. image:: images/combining_sources.*
+.. image:: images/source_values/combining_sources.*
     :align: center
 
 In this example, the LED is lit only if both buttons are pressed (like an
 `AND`_ gate):
 
+.. image:: images/source_values/combining_sources_led_buttons.*
+    :align: center
+
 .. literalinclude:: examples/combining_sources.py
 
 .. _AND: https://en.wikipedia.org/wiki/AND_gate
+
+While most devices have a :attr:`~Device.value` range between 0 and 1, some have
+a range between -1 and 1 (e.g. :class:`Motor`, :class:`Servo` and
+:class:`TonalBuzzer`). Some source tools output values between -1 and 1, which
+are ideal for these devices, for example passing :func:`~tools.sin_values` in:
+
+.. image:: images/source_values/sin_values.*
+    :align: center
+
+.. literalinclude:: examples/sin_values.py
```

Comment:
Alternatively, I guess this could do:
```python
motor.source = sin_values()
servo.source = motor
buzzer.source = servo
```
which means the sin_values only get calculated once, not three times :slightly_smiling_face:

Reply:
Yes, that's true, but I thought it would make the diagram more confusing when the point I'm trying to make is that each device is being set by `sin_values()`. I'll add a note to point that out though.

Reply:
Actually, it might be better if I do:

```python
sin = sin_values()
buzzer.source = sin
servo.source = sin
motor.source = sin
```

That means they stay in sync and the diagram can still be simple.

Reply:
No - maybe that won't work...

Reply:
I'll settle for:

```python
motor.source = sin_values()
servo.source = motor
buzzer.source = motor
```

Reply:
@waveform80 I've always wondered... given the alternatives of
```python
motor.source = sin_values()
servo.source = motor
buzzer.source = servo
```
or
```python
motor.source = sin_values()
servo.source = motor
buzzer.source = motor
```

is one version more preferable or more efficient than the other?